contracts/zero-ex/contracts/src/features/OtcOrdersFeature.sol

```diff
@@ -0,0 +1,443 @@
+            require(
+                order.makerToken == WETH,
+                "OtcOrdersFeature/INVALID_UNWRAP_WETH"
+            );
+            // Transfer maker tokens in
+            _transferERC20Tokens(
+                order.makerToken,
+                order.maker,
+                address(this),
+                makerTokenFilledAmount
+            );
+            // Unwrap WETH
+            WETH.withdraw(makerTokenFilledAmount);
+            // Transfer ETH to taker
+            _transferEth(taker, makerTokenFilledAmount);
+        } else {
+            // Transfer maker -> taker.
+            _transferERC20Tokens(
+                order.makerToken,
+                order.maker,
+                taker,
+                makerTokenFilledAmount
+            );
+        }
+    }
+
+    /// @dev Get the order info for an OTC order.
+    /// @param order The OTC order.
+    /// @return orderInfo Info about the order.
+    function getOtcOrderInfo(LibNativeOrder.OtcOrder memory order)
+        public
+        override
+        view
+        returns (LibNativeOrder.OtcOrderInfo memory orderInfo)
+    {
+        // compute order hash.
+        orderInfo.orderHash = getOtcOrderHash(order);
+
+        LibOtcOrdersStorage.Storage storage stor =
+            LibOtcOrdersStorage.getStorage();
+
+        // check tx origin nonce
+        uint256 lastNonce = stor.txOriginNonces
+            [order.txOrigin]
+            [order.txOriginNonceBucket];
+
+        if (order.txOriginNonce <= lastNonce) {
+            orderInfo.status = LibNativeOrder.OrderStatus.INVALID;
+            return orderInfo;
+        }
+
+        // Check for expiration.
+        if (order.expiry <= uint64(block.timestamp)) {
+            orderInfo.status = LibNativeOrder.OrderStatus.EXPIRED;
+            return orderInfo;
+        }
+
+        orderInfo.status = LibNativeOrder.OrderStatus.FILLABLE;
+        return orderInfo;
+    }
+
+    /// @dev Get the canonical hash of an OTC order.
+    /// @param order The OTC order.
+    /// @return orderHash The order hash.
+    function getOtcOrderHash(LibNativeOrder.OtcOrder memory order)
+        public
+        override
+        view
+        returns (bytes32 orderHash)
+    {
+        return _getEIP712Hash(
+            LibNativeOrder.getOtcOrderStructHash(order)
+        );
+    }
+
+    /// @dev Get the minimum valid nonce for a particular
+    ///      tx.origin address and nonce bucket.
+    /// @param txOrigin The address.
+    /// @param nonceBucket The nonce bucket index.
+    /// @return minNonce The minimum valid nonce value.
+    function minTxOriginNonce(address txOrigin, uint256 nonceBucket)
+        public
+        override
+        view
+        returns (uint256 minNonce)
+    {
+        LibOtcOrdersStorage.Storage storage stor =
+            LibOtcOrdersStorage.getStorage();
+        uint256 lastNonce = stor.txOriginNonces
+            [txOrigin]
+            [nonceBucket];
+        return lastNonce.safeAdd(1);
```

Comment:
What happens if this overflows?

Reply:
it'll revert. I'll change this function to return `lastNonce` instead 👍